### ISSUE TITLE
error when pip install -e .

### ISSUE THREAD

**Issue by olokojoh** [mit-han-lab/nunchaku#18]
im using A6000 48GB, got error after pip install -e .

**env info**
nvcc --version:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Mon_Apr__3_17:16:06_PDT_2023
Cuda compilation tools, release 12.1, V12.1.105
Build cuda_12.1.r12.1/compiler.32688072_0

nvidia-smi:
Tue Nov 12 09:59:00 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.98                 Driver Version: 535.98       CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA RTX A6000               On  | 00000000:40:00.0 Off |                  Off |
| 30%   26C    P8              32W / 300W |      3MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+

**torch version**
open-clip-torch                   2.20.0
pytorch-lightning                 1.9.0
pytorch-triton                    3.1.0+cf34004b8a
torch                             2.4.1+cu121
torchaudio                        2.4.1+cu121
torchdiffeq                       0.2.3
torchmetrics                      1.0.2
torchsde                          0.2.5
torchvision                       0.19.1+cu121

**error info**
Obtaining file:///root/nunchaku
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [19 lines of output]
      Traceback (most recent call last):
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 132, in get_requires_for_build_editable
          return hook(config_settings)
        File "/tmp/pip-build-env-05kx5lnd/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 483, in get_requires_for_build_editable
          return self.get_requires_for_build_wheel(config_settings)
        File "/tmp/pip-build-env-05kx5lnd/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
        File "/tmp/pip-build-env-05kx5lnd/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-05kx5lnd/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
        File "/tmp/pip-build-env-05kx5lnd/overlay/lib/python3.10/site-packages/torch/__init__.py", line 367, in <module>
          from torch._C import *  # noqa: F403
      ImportError: /tmp/pip-build-env-05kx5lnd/overlay/lib/python3.10/site-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkComplete_12_4, version libnvJitLink.so.12
      [end of output]
  
  note: This error originates from a s

**Comment by icech** [comment]
same error!

**Comment by lmxyy** [comment]
@sxtyzhangzk Could you take a look?

**Comment by xchenriang** [comment]
same error!

**Comment by hanhanpp** [comment]
+1

**Comment by Lenan22** [comment]
same error !

**Comment by weilu4606** [comment]
at A6000, same error

**Comment by sxtyzhangzk** [comment]
Hi, please try upgrading CUDA to 12.2+ (on Linux) or 12.6+ (on Windows). 
If the issue persists, please try to run 
```
python setup.py develop 
```
instead of `pip install -e .` and paste the full output here.

**Comment by nitinmukesh** [comment]
Hello @sxtyzhangzk 

When you say upgrading CUDA to 12.6+ (on Windows).

Does it means

```
C:\Users\>nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Thu_Sep_12_02:55:00_Pacific_Daylight_Time_2024
Cuda compilation tools, release 12.6, V12.6.77
Build cuda_12.6.r12.6/compiler.34841621_0
```

I have the above installed still it failed to build, I had to use the wheel to install.

**Comment by ziyanxzy** [comment]
> Hello @sxtyzhangzk
> 
> When you say upgrading CUDA to 12.6+ (on Windows).
> 
> Does it means
> 
> ```
> C:\Users\>nvcc --version
> nvcc: NVIDIA (R) Cuda compiler driver
> Copyright (c) 2005-2024 NVIDIA Corporation
> Built on Thu_Sep_12_02:55:00_Pacific_Daylight_Time_2024
> Cuda compilation tools, release 12.6, V12.6.77
> Build cuda_12.6.r12.6/compiler.34841621_0
> ```
> 
> I have the above installed still it failed to build, I had to use the wheel to install.

which wheel？

**Comment by olokojoh** [comment]
> Hello @sxtyzhangzk
> 
> When you say upgrading CUDA to 12.6+ (on Windows).
> 
> Does it means
> 
> ```
> C:\Users\>nvcc --version
> nvcc: NVIDIA (R) Cuda compiler driver
> Copyright (c) 2005-2024 NVIDIA Corporation
> Built on Thu_Sep_12_02:55:00_Pacific_Daylight_Time_2024
> Cuda compilation tools, release 12.6, V12.6.77
> Build cuda_12.6.r12.6/compiler.34841621_0
> ```
> 
> I have the above installed still it failed to build, I had to use the wheel to install.

u may try cuda 12.4. It works on my env.

**Comment by vgabbo** [comment]
I need your help, I've tried many different things but can't make it work. 
I have got Ubuntu 24.04.01 LTS, 

My nvcc is this:

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Tue_Oct_29_23:50:19_PDT_2024
Cuda compilation tools, release 12.6, V12.6.85
Build cuda_12.6.r12.6/compiler.35059454_0

And the nvidia-smi prints:
 NVIDIA-SMI 550.120                Driver Version: 550.120        CUDA Version: 12.4  
 
 I have got a rtx 4060ti. 
 When I do pip install -e . -V, it arrives till building nunchaku, but then it simply crashes and restarts the pc.
 
 I can't install cuda 12.4 because from the nvidia website I see only for ubuntu 22.04 and before, but I don't see it for 24.04.
 
 Anyone else has got the same problem?
